TestResultSummaryService/routes/getDeepSmithDataCompare.js

```diff
@@ -0,0 +1,20 @@
+const JenkinsInfo = require( '../JenkinsInfo' );
+const {removeTimeStamp} = require( './utils/removeTimeStamp' );
+const {applyDeepSmithMatch} = require( './utils/applyDeepSmithMatch' );
+
+module.exports = async ( req, res ) => {
+    const { url, buildName1, buildNum1, buildName2, buildNum2 } = req.query;
```

Comment:
- Please add url2 as well.

if (!urls2) {
 url2 = url1
}
The logic is the same for buildName2

- Please add validation. url1, buildName1, buildNum1 and buildNum2 cannot be undefined.